src/linear_algebra/quotient.lean

```diff
@@ -123,6 +124,47 @@ begin
   simpa using not_mem_s
 end
 
+/-- More general, but harder to infer, instance than `submodule.quotient.module`. -/
+instance submodule.quotient.module' (R : Type*) {A : Type*} [ring A] [comm_ring R]
+  [algebra R A] [module A M] (P : submodule A M) :
+  module R P.quotient :=
+module.of_core
+{ smul := λ c x, algebra_map R A c • x, -- Note that this has to be defeq to `c • x` if `A = R`
+  smul_add := λ c x y, smul_add _ _ _,
+  add_smul := λ c c' x, by simp only [ring_hom.map_add, add_smul],
+  mul_smul := λ c c' x, by simp only [ring_hom.map_mul, mul_action.mul_smul],
+  one_smul := λ x, by simp only [ring_hom.map_one, one_smul] }
+
+@[simp] lemma smul_mk {R A : Type*} [comm_ring R] [ring A]
```

Comment:
This lemma clashes in a surprising way with `mk_smul`. Can you just generalize the instance we already have above?